packages/app/src/app/pages/Sandbox/Editor/Workspace/WorkspaceItem/index.tsx

```diff
@@ -19,35 +19,21 @@ type Props = {
   style?: React.CSSProperties;
 };
 
-type State = {
-  open: boolean;
-};
-
-export default class WorkspaceItem extends React.Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      open: Boolean(props.defaultOpen),
-    };
-  }
-
-  shouldComponentUpdate(nextProps: Props, nextState: State) {
-    return (
-      nextState.open !== this.state.open ||
-      nextProps.disabled !== this.props.disabled ||
-      this.props.children !== nextProps.children
-    );
-  }
-
-  toggleOpen = () => this.setState({ open: !this.state.open });
-
-  render() {
-    const { children, title, keepState, disabled, actions, style } = this.props;
-    const { open } = this.state;
+const WorkspaceItem = React.memo(
+  ({
+    children,
+    title,
+    keepState,
+    disabled,
+    actions,
+    style,
+    defaultOpen,
+  }: Props) => {
+    const [open, setOpen] = useState(Boolean(defaultOpen));
 
     return (
       <>
-        <ItemHeader style={style} onClick={this.toggleOpen}>
+        <ItemHeader style={style} onClick={() => setOpen(!open)}>
```

Comment:
@SaraVieira I think you should use a function callback for using the old `open` value.
Or maybe extract it back into a `toggleOpen` function? 🤔

```suggestion
        <ItemHeader style={style} onClick={() => setOpen(isOpen => !isOpen)}>
```

Reply:
I am. gonna keep it as I think this is the most straight forward way for new comers

Reply:
@SaraVieira This can have bugs, since React can group multiple state updates